src/modules/user/user.usecase.ts

```diff
@@ -698,6 +712,84 @@ export class UserUseCases {
     });
   }
 
+  async sendAccountUnblockEmail(email: User['email']): Promise<void> {
+    const secret = this.configService.get('secrets.jwt');
+    const user = await this.userRepository.findByEmail(email);
+
+    if (!user) {
+      return;
+    }
+
+    const [mailLimit] = await this.mailLimitRepository.findOrCreate(
+      { userId: user.id, mailType: MailTypes.UnblockAccount },
+      {
+        attemptsCount: 0,
+        attemptsLimit: 5,
+      },
+    );
+
+    if (mailLimit.isLimitForTodayReached()) {
+      throw new MailLimitReachedException();
+    }
+
+    const defaultIat = getTokenDefaultIat();
+    const unblockAccountToken = SignWithCustomDuration(
+      {
+        payload: {
+          uuid: user.uuid,
+          email: user.email,
+          action: AccountTokenAction.Unblock,
+        },
+        iat: defaultIat,
+      },
+      secret,
+      '48h',
+    );
+
+    const driveWebUrl = this.configService.get('clients.drive.web');
+    const url = `${driveWebUrl}/blocked-account/${unblockAccountToken}`;
+    await this.mailerService.sendAutoAccountUnblockEmail(user.email, url);
+
+    const lastMailSentDate = Time.convertTimestampToDate(defaultIat);
+    mailLimit.increaseTodayAttemps(lastMailSentDate);
+
+    await this.mailLimitRepository.updateByUserIdAndMailType(
+      user.id,
+      MailTypes.UnblockAccount,
+      mailLimit,
+    );
+  }
+
+  async unblockAccount(
+    userUuid: User['uuid'],
+    tokenIat: number,
+  ): Promise<void> {
+    const user = await this.userRepository.findByUuid(userUuid);
+
+    if (!user) {
+      throw new BadRequestException();
+    }
+
+    const mailLimit = await this.mailLimitRepository.findByUserIdAndMailType(
+      user.id,
+      MailTypes.UnblockAccount,
+    );
+
+    const tokenIssuedAtDate = Time.convertTimestampToDate(tokenIat);
+
+    if (
+      mailLimit.lastMailSent > tokenIssuedAtDate ||
+      user.lastPasswordChangedAt > tokenIssuedAtDate
+    ) {
+      throw new ForbiddenException();
+    }
```

Comment:
I decided to not consider valid tokens if they were issued before `lastPasswordChangedAt`  basically to expire the JWT once it has been used. 

Otherwise, you can keep using the same JWT as long as it's within 48 hours and the `lastMailSent` <= token's issued date.